sql/schema/schema.go

```diff
@@ -780,3 +780,20 @@ func UnderlyingType(t Type) Type {
 	}
 	return t
 }
+
+// IsType reports if t1 is the same as t2 or if t1 is an alias of t2.
+func IsType(t1, t2 Type) bool {
+	// Comparing might panic due to mismatch types.
+	defer func() { recover() }()
+	if t1 == nil || t2 == nil {
+		return t1 == t1
+	}
+	t1, t2 = UnderlyingType(t1), UnderlyingType(t2)
+	if t1 == t2 {
```

Comment:
If we compare the underlying types, the logic is not the same anymore (as dependsOnT).